### PR TITLE
feat(textbox): add field features and char count

### LINKED ISSUE #1921: field: add helper text sections
# Feature Request

## Description

Add helper text sections in `field`.

## Context

This is the ebay ui companion to the Skin issue: https://github.com/eBay/skin/issues/2047

## Screenshots

image

### PR DESCRIPTION
**Pull request by LuLaValva**
- Resolves #1921 

## Description

- Add `description`, `label`, and `char-count` attribute tags to `<ebay-textbox>`
- Automatically wrap textbox with `field` when required

## Screenshots

image
image
image
image
image

### CODE REVIEW

File: src/components/ebay-textbox/component.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+    }
+
+    updateCharCount(el) {
+        // debounce
+        clearTimeout(this._charCountTimeout);
+        this._charCountTimeout = setTimeout(() => {
+            this.state.charCount = el.value.length;
+            this.state.value = el.value;
+            this.emit("char-count", {
```
Comment (agliga) [inline src/components/ebay-textbox/component.js:51]: I don't like this event name. Maybe `on-char-count-change` or `on-char-change`

Comment (ArtBlue) [reply src/components/ebay-textbox/component.js:51]: I second this suggestion. Shouldn't emits of all events be preceded by `on-` at the very least?

Comment (ArtBlue) [reply src/components/ebay-textbox/component.js:51]: We already do that in other components. 
image

Comment (LuLaValva) [reply src/components/ebay-textbox/component.js:51]: That component still emits `blur`, `click`, and `focus`. We just change the name in storybook.

File: src/components/ebay-textbox/component.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    focus() {
+        this.getEl("input").focus();
+    }
+
+    _setupMakeup() {
+        // TODO: makeup-floating-label should be updated so that we can remove the event listeners.
+        // It probably makes more sense to just move this functionality into Marko though.
+        if (this.input.floatingLabel) {
+            if (this._floatingLabel) {
```
[the remark this answers is not in the document]
Comment (LuLaValva) [reply src/components/ebay-textbox/component.js:28]: This is not something that I built, I pulled it directly from the previous `component-browser.json`. I'll look further into this before it's merged.

File: src/components/ebay-textbox/test/__snapshots__/renders-a-textarea-element-with-postfix-icon-button.expected.html [hunk 3]
```diff
@@ -29,6 +29,8 @@
       [36m/>[39m
     [36m</svg>[39m
     [36m<input[39m
+      [33maria-describedby[39m=[32m"s0-@textbox-description s0-@textbox-char-count"[39m
+      [33maria-live[39m=[32m"off"[39m
```
[the remark this answers is not in the document]
Comment (LuLaValva) [reply src/components/ebay-textbox/test/__snapshots__/renders-a-textarea-element-with-postfix-icon-button.expected.html:33]: Good catch, I'll prevent that from happening.

**Review by LuLaValva — COMMENTED** [review]
Thanks for the comments!

File: src/components/ebay-textbox/component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-textbox/component.js:28]: Lets not mess with this. We've had some issues before with this

File: src/components/ebay-textbox/index.marko [hunk 4]
```diff
@@ -17,7 +17,10 @@ static var ignoredAttributes = [
     "postfixIcon",
     "inputSize",
     "opaqueLabel",
-    "toJSON"
+    "toJSON",
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-textbox/index.marko]: I think this is a server side render and rerender client issue
We don't do server rendering in storybook.

File: src/components/ebay-textbox/component.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArtBlue) [reply src/components/ebay-textbox/component.js:28]: Yep, it's fine to leave it alone for now; just commenting about the practice in general. We can revisit this at another time.

**Comment by LuLaValva** [comment]
- Closing in favor of #1965

**Pull request closed without merging**